powerscale/provider/snapshot_datasource.go

```diff
@@ -238,12 +278,31 @@ func (d *SnapshotDataSource) Read(ctx context.Context, req datasource.ReadReques
 			)
 			return
 		}
+	}
+
+	if plan.SnapshotFilter != nil && plan.SnapshotFilter.Name.ValueString() != "" {
+		for _, sdm := range fulldetail {
+			if plan.SnapshotFilter.Name.ValueString() == sdm.Name.ValueString() {
+				state.Snapshots = append(state.Snapshots, sdm)
+			}
+		}
+		// If after the filter the length is still zero then that filter is invalid
+		if len(state.Snapshots) == 0 {
+			resp.Diagnostics.AddError(
+				"Error getting snapshots",
+				fmt.Sprintf("Name `%s` is invalid, it has no snapshots ", plan.SnapshotFilter.Path.ValueString()),
```

Comment:
I think the better message would be "Unable to find snapshot with name %s", plan.SnapshotFilter.Name.ValueString"

Reply:
Changed this in the recent commit